OneSignalSDK/onesignal/core/src/main/java/com/onesignal/notifications/INotificationLifecycleListener.kt

```diff
@@ -0,0 +1,29 @@
+package com.onesignal.notifications
+
+/**
+ * Implement this interface and provide an instance to [INotificationsManager.addForegroundLifecycleListener]
+ * in order to receive control during the lifecycle of the notification. This handler will *only* receive
+ * control when the application is in the foreground when the notification is received.
+ *
+ * @see [Foreground Notification Received Event | OneSignal Docs](https://documentation.onesignal.com/docs/sdk-notification-event-handlers#foreground-notification-received-event)
```

Comment:
This should use our v11 docs which are here: https://dash.readme.com/project/onesignal/v11.0/docs/sdk-notification-event-handlers

Note that no one has provided a thorough review of these docs, and they may be wildly inaccurate or even incomplete.